src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -42,82 +62,96 @@ internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder,
-                .AddFilter<OpenTelemetryLoggerProvider>("Function.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
-                .AddFilter<OpenTelemetryLoggerProvider>("Azure.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
-                // Host.Results and Host.Aggregator are used to emit metrics, ignoring these categories.
-                .AddFilter<OpenTelemetryLoggerProvider>("Host.Results", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
-                .AddFilter<OpenTelemetryLoggerProvider>("Host.Aggregator", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
-                // Ignoring all Microsoft.Azure.WebJobs.* logs like /getScriptTag and /lock.
-                .AddFilter<OpenTelemetryLoggerProvider>("Microsoft.Azure.WebJobs.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled);
+                .AddDefaultOpenTelemetryFilters();
 
             // Azure SDK instrumentation is experimental.
             AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
 
+            ConfigureTracing(loggingBuilder, enableOtlp, enableAzureMonitor, azMonConnectionString, credential);
+
+            ConfigureEventLogLevel(loggingBuilder, context.Configuration);
+        }
+
+        private static void ConfigureTracing(ILoggingBuilder loggingBuilder, bool enableOtlp, bool enableAzureMonitor, string azMonConnectionString, TokenCredential credential)
+        {
             loggingBuilder.Services.AddOpenTelemetry()
                 .ConfigureResource(r => ConfigureResource(r))
-                .WithTracing(b =>
+                .WithTracing(builder =>
                 {
-                    b.AddSource("Azure.*");
-                    b.AddAspNetCoreInstrumentation();
-                    b.AddHttpClientInstrumentation(o =>
-                    {
-                        o.FilterHttpRequestMessage = _ =>
-                        {
-                            Activity activity = Activity.Current?.Parent;
-                            return activity == null || !activity.Source.Name.Equals("Azure.Core.Http");
-                        };
-                    });
+                    builder.AddSource("Azure.*")
+                           .AddAspNetCoreInstrumentation()
+                           .AddHttpClientInstrumentation(o =>
+                           {
+                               o.FilterHttpRequestMessage = static (httpRequestMessage) =>
+                               {
+                                   if (httpRequestMessage.RequestUri?.AbsoluteUri is not { Length: > 0 } uri)
+                                   {
+                                       return false;
+                                   }
+
+                                   foreach (string substring in ExcludedRequestSubstrings)
+                                   {
+                                       if (uri.IndexOf(substring, StringComparison.Ordinal) >= 0)
```

Comment:
I think this will need to be made more robust:

1. This looks like it might not be cheap to perform on a hotpath like this
2. These substrings may have a chance of false-positives.

Possible solutions:

1. Is there something besides URI on the `HttpRequestMessage` we can use to identify these calls?
2. Can we _add_ something to the `HttpRequestMessage` to identify these calls? Such as adding a value to `.Options` to indicate this call is to be filtered out.
3. Can we use an AsyncLocal to suppress Http client telemetry collection?

Reply:
Removed HTTPClient instrumentation. I don't see much value in having HTTPClient instrumentation on the host. It captures all internal calls, including those made to AppInsights ingestion endpoint (Az Mon Exporter) and by Azure SDKs like Storage, which already provide distributed traces. 
Removing this would eliminate the need for filtering out unnecessary traces.